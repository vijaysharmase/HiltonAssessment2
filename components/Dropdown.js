import React from "react";
import styled from "styled-components";

const StyledDropdown = styled.select`
	font: 0.8em "Montserrat", sans-serif;
	height: 2em;
	width: 3em;
`;

const Dropdown = ({ ...props }) => <StyledDropdown {...props} />;

export default Dropdown;
