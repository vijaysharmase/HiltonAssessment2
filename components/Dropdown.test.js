import React from "react";
import enzyme from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Dropdown from "./Dropdown";
enzyme.configure({ adapter: new Adapter() });

describe("Checkbox component", () => {
	var value = "2";
	const props = {
		className: "new-checkbox",
		checked: true,
		disabled: false,
		value: "2",
		onChange: event => {
			value = event.target.value;
		}
	};

	it("check dropdown change functionality", () => {
		const wrapper = enzyme.mount(
			<Dropdown {...props}>
				<option value="1">1</option>
				<option value="2">2</option>
			</Dropdown>
		);
		wrapper.find("select").simulate("change", { target: { value: "1" } });
		expect(value).toBe("1");
	});

	it("Compare snapshots", () => {
		const wrapper = renderer.create(
			<Dropdown {...props}>
				<option value="1">1</option>
				<option value="2">2</option>
			</Dropdown>
		);
		const dropdown = wrapper.toJSON();
		expect(dropdown).toMatchSnapshot();
	});
});
