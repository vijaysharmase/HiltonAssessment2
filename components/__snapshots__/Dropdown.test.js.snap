// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Compare snapshots 1`] = `
<select
  checked={true}
  className="new-checkbox sc-bdVaJa eUnrpt"
  disabled={false}
  onChange={[Function]}
  value="2"
>
  <option
    value="1"
  >
    1
  </option>
  <option
    value="2"
  >
    2
  </option>
</select>
`;
